gokce_fingerprint_verification.py

```diff
@@ -0,0 +1,184 @@
+    """
+    for i in range(len(list_2d)):
+        if list_2d[i].count(value) >= count:
+            return i
+
+# Function to check original and variant fingerprints
+def simple_and_variant_check(filename1, filename2, pattern_value, pattern_count, rows_to_compare=None):
+    """
+    Compare two 2D lists and return the result.
+    """
+
+
+    #  Get 2d Lists from files
+    list1 = read_and_list_fprint(filename1)
+    list2 = read_and_list_fprint(filename2)
+
+
+
+    # Get the top of the fingerprint
+    start_row1 = find_row_with_pattern(list1, pattern_value, pattern_count)
+    start_row2 = find_row_with_pattern(list2, pattern_value, pattern_count)
+    
+
+    # Find the end of the fingerprint
+    end_row1 = start_row1 + rows_to_compare
+    end_row2 = start_row2 + rows_to_compare
+
+
+
+    total_elements = 0
+    matching_elements = 0
+    #compares the two lists against eachother
+    for row1, row2 in zip(list1[start_row1:end_row1], list2[start_row2:end_row2]):
+        # adds 1 to total elements at every step
+        for item1, item2 in zip(row1, row2):
+            total_elements += 1
+            # adds 1 basen on matching element
+            if item1 == item2:
+                matching_elements += 1 
+            else: 0
+
+            
+        
+
+    # Calculates similarity percentage and prints it as string
+    overall_similarity = (matching_elements / total_elements) * 100
+    overall_similarity_str = f"{overall_similarity:.2f}%"
+
+    # Returns different outcomes based on the percentage amount.
+    if overall_similarity == 100.0:
+        return "Identical Print. Match Percentage:", overall_similarity_str
+    elif overall_similarity <= 75.0:
+        return "Not Similar, cannot be the same person. Match Percentage:", overall_similarity_str
+    elif 75.0 <= overall_similarity <= 88.0: 
+        return "Similar but not enough to determine if same person or not. Match Percentage:", overall_similarity_str
+    elif 88.0 <= overall_similarity <= 93.0: 
+        return "Very similar, most likely the same person. Match Percentage:", overall_similarity_str
+    else:
+        return "Definitely the same person. Match Percentage:", overall_similarity_str
+
+
+
+def shifted_check(filename1, filename2, pattern_value, pattern_count, rows_to_compare=None, max_shift=None):
+    """ Compare a shifted fingerprint to another shifted or a non shifted one """
+
+
+        #  Get 2d Lists from files
+    list1 = read_and_list_fprint(filename1)
+    list2 = read_and_list_fprint(filename2)
+
+
+
+    # Get the top of the fingerprint
+    start_row1 = find_row_with_pattern(list1, pattern_value, pattern_count)
+    start_row2 = find_row_with_pattern(list2, pattern_value, pattern_count)
+    
+
+    # Find the end of the fingerprint
+    end_row1 = start_row1 + rows_to_compare
+    end_row2 = start_row2 + rows_to_compare
+
+
+
+    total_elements = sum(len(row) for row in list1)
```

Comment:
This would be including all of the early metadata that isn't part of the fingerprint itself.